### PR TITLE
Non-Version Specific changes for Future NuttX upgades

### PR DESCRIPTION
**Pull request by davids5**
In the 10.1.0 Of Nuttx this changes will be needed. This PR preps the current PX4 code base for theses structural changes. 

Using inttypes - will be enforced in 10.1.0 as will type checking.

getops interface will be Thread safe and accesors will be macros that collide the cli.getops interface. The change here in uses mixed case to avoid the collisions, 

STM32 Timers will be based on the Basic timer defines and advanced

Includes for debug/syslog/assert/schedule  will only be public API and not layerd.

### CODE REVIEW

**Comment by davids5** [comment]
@dagar dropped the cmake change - Please merge this

**Comment by davids5** [comment]
Re-based on current master

**Comment by davids5** [comment]
Re-based on current master

File: src/drivers/pwm_out/PWMOut.cpp [hunk 1]
```diff
@@ -381,7 +381,7 @@ int PWMOut::set_mode(Mode mode)
  *                                    For Oneshot there is no rate, 0 is therefore used
  *                                    to  select Oneshot mode
  */
-int PWMOut::set_pwm_rate(uint32_t rate_map, unsigned default_rate, unsigned alt_rate)
+int PWMOut::set_pwm_rate(unsigned rate_map, unsigned default_rate, unsigned alt_rate)
```
[the remark this answers is not in the document]
Comment (davids5) [reply src/drivers/pwm_out/PWMOut.cpp:384]: There should be none. The type that have size requirements are explicit and the PRIxxx macros are used to give better warnings

[the remark this answers is not in the document]
Comment (JacobCrabill) [reply src/drivers/pwm_out/PWMOut.cpp:384]: Still, for my understanding, is it correct that (for example) the return type of many system calls (those that return `int`) has changed size with the NuttX update?

[the remark this answers is not in the document]
Comment (davids5) [reply src/drivers/pwm_out/PWMOut.cpp:384]: no

**Pull request merged**